src/ray/core_worker/core_worker.cc

```diff
@@ -1174,8 +1174,77 @@ Status CoreWorker::SetResource(const std::string &resource_name, const double ca
   return local_raylet_client_->SetResource(resource_name, capacity, client_id);
 }
 
-Status CoreWorker::ForceSpillObjects(const std::vector<ObjectID> &object_ids) {
-  return local_raylet_client_->ForceSpillObjects(object_ids);
+Status CoreWorker::SpillObjects(const std::vector<ObjectID> &object_ids) {
+  absl::Mutex mutex;
+  auto num_remaining = std::make_shared<size_t>(object_ids.size());
+  auto final_status = std::make_shared<Status>();
+  auto ready_promise = std::make_shared<std::promise<void>>(std::promise<void>());
+
+  for (const auto &object_id : object_ids) {
+    RAY_LOG(DEBUG) << "Requesting spill for object " << object_id;
+    AddLocalReference(object_id, "<temporary (get object status)>");
+
+    rpc::Address owner_address;
+    auto has_owner = reference_counter_->GetOwner(object_id, &owner_address);
```

Comment:
Can you break this up into some helper functions? Also, I don't think we need to be so rigorous to check error conditions for what is essentially a debug / benchmark utility.

It would be a lot simpler to simply try to resolve the owner and fire off requests without any error checking, maybe just log a warning.

Reply:
Hmm I do think it's nice that the utility is synchronous right now. I think it's worthwhile to try and make this work since the utility itself is useful, not just for debugging and benchmarking.

Which specific part do you want to break out?

Reply:
I agree it's good to keep it synchronous.

My observation is just that this function is too big and is difficult to read.